tools/sigh.ts

```diff
@@ -137,25 +144,40 @@ function targetIsUpToDate(relativeTarget: string, relativeDeps: string[]): boole
     }
   }
 
-  console.log(`Skipping step; '${relativeTarget}' is up-to-date`);
+  if (!quiet) {
+    console.log(`Skipping step; '${relativeTarget}' is up-to-date`);
+  }
   return true;
 }
 
 
 function check(): boolean {
   const nodeRequiredVersion = require('../package.json').engines.node;
   const npmRequiredVersion = require('../package.json').engines.npm;
+  const emsdkRequiredVersion = require('../package.json').engines.emsdk;
 
   if (!semver.satisfies(process.version, nodeRequiredVersion)) {
     throw new Error(`at least node ${nodeRequiredVersion} is required, you have ${process.version}`);
   }
 
-  const npmCmd = saneSpawnWithOutput('npm', ['-v']);
-  const npmVersion = String(npmCmd.stdout);
+  const npmVersion = saneSpawnWithOutput('npm', ['-v']).stdout;
   if (!semver.satisfies(npmVersion, npmRequiredVersion)) {
     throw new Error(`at least npm ${npmRequiredVersion} is required, you have ${npmVersion}`);
   }
 
+  const emsdkResult = saneSpawnWithOutput('npx', ['emsdk-run', 'em++', '--version']);
+  const match = emsdkResult.stdout.match(/\nemcc [^0-9.]+ (X[0-9.]+) /);
```

Comment:
Might be nice to have a comment on the regex, very hard to tell what it's doing otherwise.

Reply:
Good point, will update.